tests/modeling/test_rpn.py

```diff
@@ -77,6 +79,49 @@ def test_rpn(self):
             )
             self.assertTrue(torch.allclose(proposal.objectness_logits, expected_objectness_logit))
 
+    @unittest.skipIf(TORCH_VERSION < (1, 6), "Insufficient pytorch version")
+    def test_rpn_scriptability(self):
+        torch.manual_seed(121)
+        cfg = get_cfg()
+        cfg.MODEL.PROPOSAL_GENERATOR.NAME = "RPN"
+        cfg.MODEL.ANCHOR_GENERATOR.NAME = "DefaultAnchorGenerator"
+        cfg.MODEL.RPN.BBOX_REG_WEIGHTS = (1, 1, 1, 1)
+        backbone = build_backbone(cfg)
+        proposal_generator = build_proposal_generator(cfg, backbone.output_shape()).eval()
+        num_images = 2
+        images_tensor = torch.rand(num_images, 20, 30)
+        image_sizes = [(10, 10), (20, 30)]
+        images = ImageList(images_tensor, image_sizes)
+        image_shape = (15, 15)
+        num_channels = 1024
+        features = {"res4": torch.rand(num_images, num_channels, 1, 2)}
+
+        fields = {"proposal_boxes": "Boxes", "objectness_logits": "Tensor"}
+        proposal_generator_script, new_instance = export_torchscript_with_patch_instance(
+            proposal_generator, fields
+        )
+
+        box_data = [[1, 1, 3, 3], [2, 2, 6, 6]]
+        all_gt_instances = []
+        for t in box_data:
+            gt_boxes = torch.tensor([t], dtype=torch.float32)
+            gt_instance = new_instance(image_shape)
+            gt_instance.gt_boxes = Boxes(gt_boxes)
+            all_gt_instances.append(gt_instance)
+
+        with EventStorage():  # capture events in a new storage to discard them
```

Comment:
we should run test in eval mode only, and in that case this is not needed

Reply:
done